### PR TITLE
added virtualization for the conversation list

### PR DESCRIPTION
**Pull request by bhavya2611**
Added react virtuoso for virtualization

#243

### CODE REVIEW

**Comment by daria-github** [comment]
Closing for the time being — encountered several issues during testing.

File: component-library/components/MessagePreviewCard/MessagePreviewCard.tsx [hunk 1]
```diff
@@ -56,12 +56,8 @@ export const MessagePreviewCard = ({
   isSelected,
   conversationDomain,
 }: MessagePreviewCard) => {
```
Comment (daria-github) [inline component-library/components/MessagePreviewCard/MessagePreviewCard.tsx:58]: why remove this logic?

File: hooks/useListConversations.tsx [hunk 2]
```diff
@@ -55,7 +55,10 @@ export const useListConversations = () => {
       setPreviewMessages(newPreviewMessages);
 
       for (const convo of allConversations) {
-        if (convo.peerAddress !== walletAddress) {
+        if (
+          convo.peerAddress !== walletAddress &&
+          newPreviewMessages.has(getConversationId(convo))
+        ) {
```
Comment (rygine) [inline hooks/useListConversations.tsx:61]: is this to prevent showing a conversation without a message?

Comment (bhavya2611) [reply hooks/useListConversations.tsx:61]: Yes

File: component-library/components/MessagePreviewCard/MessagePreviewCard.tsx (hunk 1 above)
Comment (bhavya2611) [reply component-library/components/MessagePreviewCard/MessagePreviewCard.tsx:58]: My bad, this is a regression, will add this back. I got confused with the feedback convo feature and this

File: hooks/useListConversations.tsx [hunk 3]
```diff
@@ -43,22 +43,18 @@ export const useListConversations = () => {
...
-          newPreviewMessages.set(preview.key, preview.message);
-        }
-      }
       setPreviewMessages(newPreviewMessages);
-
-      for (const convo of allConversations) {
-        if (convo.peerAddress !== walletAddress) {
-          conversations.set(getConversationId(convo), convo);
-        }
-      }
```
[the remark this answers is not in the document]
Comment (bhavya2611) [reply hooks/useListConversations.tsx:61]: Thank you for pointing this out, I actually combined all three loops here, to improve performance, hence the logic changed.

File: component-library/components/SideNav/SideNav.tsx [hunk 4]
```diff
@@ -180,7 +178,7 @@ const SideNav = ({
                     <span className="font-bold" data-testid="wallet-address">
                       {shortAddress(displayAddress ?? "")}
                     </span>
-                    {walletAddress && (
+                    {walletAddress && displayAddress !== walletAddress && (
```
[the remark this answers is not in the document]
Comment (bhavya2611) [reply component-library/components/SideNav/SideNav.tsx:181]: yes, this was an existing bug, we would show the address twice, now that won't happen

**Review by rygine — APPROVED** [review]

**Pull request merged**